client/shared/source-utils.js

```diff
@@ -0,0 +1,293 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+// NOTE: we'll be falling back to the environments URL global
+// const { URL } = require("sdk/url");
```

Comment:
stubbed out because browser windows have a global `URL` object that should be sufficient. This might not be the case in a testing environment or node